### PR TITLE
[ENH] MDN Improved Losses (nGem)

### PR DESCRIPTION
**Pull request by joshdunnlime**
#### Reference Issues/PRs

No Issue but referenced [here](https://github.com/sktime/skpro/pull/796#issuecomment-4027898053).

#### What does this implement/fix? Explain your changes.

Mixture density networks typically uses the negative log-likelihood (nll) objective, which can suffer from slow convergence and mode collapse. The natural gradient expectation maximization (ngem) objective can achieve up to 10x faster convergence while adding almost zero computational overhead, and scales well to high-dimensional data where nll fails.

[Learning Mixture Density via Natural Gradient Expectation Maximization](https://arxiv.org/html/2602.10602v1)

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

~General API structure for adding ngem specific learning rate. ngem typically performs better with lower lr, so adding an automated lr multiplier has been considered. Is this designed correctly?~

I have removed `ngem_lr_scaling` as it obscures true `lr` and sets a bad standard if we wish to add more objectives that also benefit different magnitudes of scaling.

#### Did you add any tests for the change?

Existing MDN tests to still pass. Add specific ngem tests also.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/skpro/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/skpro/blob/main/.all-contributorsrc) in the `skpro` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [x] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [x] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice! Though, should the log-loss - previous behaviour - not also be an option?

I think we should not break user behaviour, so I would suggest to retain previous usage of the log-loss as an option.

Or does it not get removed? Would appreciate explanation if I misunderstood.

**Comment by joshdunnlime** [comment]
No we don't remove log-loss (or rather [nll](https://github.com/sktime/skpro/pull/1042/changes#diff-289ef28ac2f44b1654b2f73e2d22c7a590e8674d5407d1a80adb3b38d82964c5R154)). `ngem` is provided as an additional option to the loss argument.

**Review by fkiraly — APPROVED** [review]
Ah, ok. You are just returning unnormalized logits from the network now, right?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Can you kindly look into the failure message? Looks related to the changes. Have you run `check_estimator`?

**Comment by joshdunnlime** [comment]
> Can you kindly look into the failure message? Looks related to the changes. Have you run `check_estimator`?

Exactly. We just pass those to the loss the users selects.

I'll fix the test ASAP. Thanks!

**Comment by fkiraly** [comment]
kind reminder

**Comment by joshdunnlime** [comment]
@fkiraly - this should be fixed now. It passes locally.

**Review by fkiraly — APPROVED** [review]
Looks good now!
